spras/evaluation.py

```diff
@@ -67,63 +67,73 @@ def load_files_from_dict(self, gold_standard_dict: Dict):
+                f"Gold standard '{self.label}': both node_files and edge_files provided. "
+                "Only one type is allowed in a gold standard dataset."
             )
-        if not has_node_file and not has_edge_file:
+        if not has_node_files and not has_edge_files:
             raise ValueError(
-                f"Gold standard '{self.label}': neither node_file nor edge_file provided."
+                f"Gold standard '{self.label}': neither node_files nor edge_files provided."
             )
 
-        # TODO: later iteration - allow for multiple node or edge files in the list
-        # The files will be merged together to make one large node or edge table
-        if has_node_file and len(node_file) != 1:
-            raise ValueError(
-                f"Gold standard '{self.label}': node_file must contain exactly one file."
-            )
-        if has_edge_file and len(edge_file) != 1:
-            raise ValueError(
-                f"Gold standard '{self.label}': edge_file must contain exactly one file."
-            )
+        if has_node_files:
 
-        if has_node_file:
-            single_node_table = pd.read_table(os.path.join(data_loc, node_file[0]), header=None)
-            if single_node_table.shape[1] != 1:
-                raise ValueError(
-                    f"Gold standard '{self.label}': the provided node_file must have exactly 1 column of nodes."
-                )
-            single_node_table.columns = [self.NODE_ID]
-            self.node_table = single_node_table
+            node_tables = []
+            for node_file in node_files:
+                node_table = pd.read_table(os.path.join(data_loc, node_file), header=None)
+                if node_table.shape[1] != 1:
+                    raise ValueError(
+                        f"Gold standard '{self.label}': each node_file provided must have exactly 1 column of nodes."
+                    )
+                node_tables.append(node_table)
 
-        if has_edge_file:
-            single_edge_table = pd.read_table(os.path.join(data_loc, edge_file[0]), header=None)
-            if single_edge_table.shape[1] != 3:
-                raise ValueError(
-                    f"Gold standard '{self.label}': the provided edge_file must have exactly 3 columns (Interactor1, Interactor2, Direction)."
-                )
-            single_edge_table.columns = ['Interactor1', 'Interactor2', 'Direction']
+            merged_node_table = pd.concat(node_tables, ignore_index=True)
+            merged_node_table = merged_node_table.drop_duplicates()
+            merged_node_table.columns = [self.NODE_ID]
+
+            self.node_table = merged_node_table
+
+
+        if has_edge_files:
+            edge_tables = []
+
+            for edge_file in edge_files:
+                edge_table = pd.read_table(os.path.join(data_loc, edge_file), header=None)
+                if edge_table.shape[1] != 3:
```

Comment:
Should directionality be optional like it is for input edge files?

Reply:
Yeah, just so we stay consistent. Any edges without directionality will be marked as undirected.